### PR TITLE
[MGDOBR-968] - Create Logger action for development

### PR DESCRIPTION
**Pull request by vaibhavjainwiz**
https://issues.redhat.com/browse/MGDOBR-968

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [ ] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [ ] The layers in the `kustomize` folder have been updated accordingly.
- [ ] All new functionality is tested
- [ ] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

File: processor/configuration/src/main/resources/schemas/actions/catalog.json [hunk 1]
```diff
@@ -40,5 +40,11 @@
     "name": "Webhook",
     "description": "Send the event to a webhook.",
     "isConnector": false
+  },
+  {
+    "id": "logger_sink_0.1",
+    "name": "logger",
+    "description": "Log event.",
+    "isConnector": false
```
Comment (r00ta) [inline processor/configuration/src/main/resources/schemas/actions/catalog.json:48]: The user of the service has no access to the executor logs, I'd not expose it in the catalog as this is just an internal debugging detail

Comment (vaibhavjainwiz) [reply processor/configuration/src/main/resources/schemas/actions/catalog.json:48]: Current implementation enforce us to define it in catalog else it give error.

Comment (manstis) [reply processor/configuration/src/main/resources/schemas/actions/catalog.json:48]: I agree; you cannot expose this _development_ Action in the usual way.

If it is not in the catalogue you may find how it is validated also needs refactoring.

tl;dr; adding a Logger Action is not as straight forward as you think.

**Review by manstis — CHANGES_REQUESTED** [review]
You cannot add a _development_ Action the way you have.

The whole mechanism of catalogue, lookup and validation needs to be changed to support this.

File: processor/configuration/src/main/resources/schemas/actions/catalog.json (hunk 1 above)
Comment (vaibhavjainwiz) [reply processor/configuration/src/main/resources/schemas/actions/catalog.json:48]: Our product not going to customer any sooner. This logger action could help developer and tester both to speed up development. 

There could be multiple ways to hide it from customer like we could create a `development_catalog.json` where we could define these type of Action. We could think of more ways.

Comment (manstis) [reply processor/configuration/src/main/resources/schemas/actions/catalog.json:48]: I agree with your reasoning; but this PR, IMO, should also include whatever "...way[s] to hide it from customer..." otherwise it will be forgotten and never addressed.

**Comment by vaibhavjainwiz** [comment]
> You cannot add a _development_ Action the way you have.
> 
> The whole mechanism of catalogue, lookup and validation needs to be changed to support this.

sorry.. I didn't get you. What need to be change?
It you are looking for test parameter then it because current implementation enforce it. It a bug in current implementation that need to be fixed.
